autogpt_platform/backend/backend/blocks/smart_decision_maker.py

```diff
@@ -130,6 +139,124 @@ def _get_tool_graph_metadata(self, node_id: str, graph: Any) -> List[Any]:
+            links: The list of links connected to the sink node.
+
+        Returns:
+            A dictionary representing the function signature in the format expected by LLM tools.
+
+        Raises:
+            ValueError: If the block specified by sink_node.block_id is not found.
+        """
+        block = get_available_blocks()[sink_node.block_id]
+        if not block:
+            raise ValueError(f"Block not found: {sink_node.block_id}")
+
+        tool_function: dict[str, Any] = {
+            "name": re.sub(r"[^a-zA-Z0-9_-]", "_", block().name).lower(),
+            "description": block().description,
+        }
+
+        properties = {}
+        required = []
+
+        for link in links:
+            sink_block_input_schema = block().input_schema
+            description = (
+                sink_block_input_schema.model_fields[link.sink_name].description
+                if link.sink_name in sink_block_input_schema.model_fields
+                and sink_block_input_schema.model_fields[link.sink_name].description
+                else f"The {link.sink_name} of the tool"
+            )
+            properties[link.sink_name.lower()] = {
+                "type": "string",
+                "description": description,
+            }
+
+        tool_function["parameters"] = {
+            "type": "object",
+            "properties": properties,
+            "required": required,
+            "additionalProperties": False,
+            "strict": True,
+        }
+
+        return {"type": "function", "function": tool_function}
+
+    @staticmethod
+    def _create_agent_function_signature(
+        sink_node: Any, links: list[Any], tool_graph_metadata: list[Any]
+    ) -> dict[str, Any]:
+        """s∫
+        Creates a function signature for an agent node.
+
+        Args:
+            sink_node: The agent node for which to create a function signature.
+            links: The list of links connected to the sink node.
+            tool_graph_metadata: List of metadata for available tool graphs.
+
+        Returns:
+            A dictionary representing the function signature in the format expected by LLM tools.
+
+        Raises:
+            ValueError: If the graph metadata for the specified graph_id and graph_version is not found.
+        """
+        graph_id = sink_node.input_default["graph_id"]
+        graph_version = sink_node.input_default["graph_version"]
+
+        sink_graph_meta = next(
+            (
+                meta
+                for meta in tool_graph_metadata
+                if meta.id == graph_id and meta.version == graph_version
+            ),
+            None,
+        )
+
+        if not sink_graph_meta:
+            raise ValueError(
+                f"Sink graph metadata not found: {graph_id} {graph_version}"
+            )
+
+        tool_function: dict[str, Any] = {
+            "name": re.sub(r"[^a-zA-Z0-9_-]", "_", sink_graph_meta.name).lower(),
```

Comment:
I see this being used in two places, is it actually fine to replace the name here ? why is it needed ?